header-only/tests/test_1494-layout-builder.cpp

```diff
@@ -1209,6 +1212,62 @@ test_Regular_size0() {
   assert(builder.length() == 0);
 }
 
+void
+test_Indexed() {
+  IndexedBuilder<uint32_t, NumpyBuilder<double>> builder;
+  assert(builder.length() == 0);
+
+  std::cout << "Executing test_Indexed";
+
+  auto& subbuilder = builder.append_index();
+  subbuilder.append(1.1);
+
+  builder.append_index();
+  subbuilder.append(2.2);
+
+  double data[3] = {3.3, 4.4, 5.5};
+
+  builder.extend_index(1);
+  subbuilder.extend(data, 3);
+
```

Comment:
@ManasviGoyal I wanted to see the test failing so I am extending the index by 1 and content by 3, surprisingly the test passes anyway...do you see what is wrong?

Reply:
@zonca That is strange. I checked and for me it failed.
```
void test_Indexed(): Assertion `builder.is_valid(error) == true'
```

Reply:
@ManasviGoyal what is worrysome is that also CI passed fine.
the only strangeness is that it prints out `Executing test_Indexed::endgroup::` but now sure what that means:

https://github.com/zonca/awkward/actions/runs/7103861309/job/19337516467?pr=3

Reply:
This makes sense. I was testing it by running the specific test file in C++ only that's why the test for failing for me.